search/tests/test_views.py

```diff
@@ -447,3 +447,84 @@ def test_search_from_url(self):
             searcher.index("courseware_content", [{"id": "FAKE_ID_3", "content": {"text": "Here comes the sun"}}])
+
+
+@override_settings(SEARCH_ENGINE="search.tests.utils.ForceRefreshElasticSearchEngine")
+@override_settings(ELASTIC_FIELD_MAPPINGS={"start_date": {"type": "date"}})
+@override_settings(COURSEWARE_INDEX_NAME=TEST_INDEX_NAME)
+class ElasticSearchUrlTest(TestCase, SearcherMixin):
+    """Elastic-specific tests"""
+    def setUp(self):
+        self.searcher.index(
+            "courseware_content",
+            [
+                {
+                    "course": "ABC/DEF/GHI",
+                    "id": "FAKE_ID_1",
+                    "content": {
+                        "text": "It seems like k-means clustering would work in this context."
+                    },
+                    "test_date": datetime(2015, 1, 1),
+                    "test_string": "ABC, It's easy as 123"
+                }
+            ]
+        )
+        self.searcher.index(
+            "courseware_content",
+            [
+                {
+                    "course": "ABC/DEF/GHI",
+                    "id": "FAKE_ID_2",
+                    "content": {
+                        "text": "It looks like k-means clustering could work in this context."
+                    }
+                }
+            ]
+        )
+        self.searcher.index(
+            "courseware_content",
+            [
+                {
+                    "course": "ABC/DEF/GHI",
+                    "id": "FAKE_ID_3",
+                    "content": {
+                        "text": "It looks like k means something different in this context."
+                    }
+                }
+            ]
+        )
+
+    def test_quoted_phrase(self):
+        code, results = post_request({"search_string": "\"in this context\""})
+        self.assertTrue(199 < code < 300)
+        self.assertEqual(results["total"], 3)
+
+        code, results = post_request({"search_string": "\"looks like\""})
+        self.assertTrue(199 < code < 300)
+        self.assertEqual(results["total"], 2)
+
+        code, results = post_request({"search_string": "\"in this context\""}, "ABC/DEF/GHI")
+        self.assertTrue(199 < code < 300)
+        self.assertEqual(results["total"], 3)
+
+        code, results = post_request({"search_string": "\"looks like\""}, "ABC/DEF/GHI")
+        self.assertTrue(199 < code < 300)
+        self.assertEqual(results["total"], 2)
+
+    def test_hyphenated_phrase(self):
+        code, results = post_request({"search_string": "k-means"})
+        self.assertTrue(199 < code < 300)
+        self.assertEqual(results["total"], 3)
+
+        code, results = post_request({"search_string": "k-means"}, "ABC/DEF/GHI")
+        self.assertTrue(199 < code < 300)
+        self.assertEqual(results["total"], 3)
+
+    def test_malformed_query_handling(self):
+        code, results = post_request({"search_string": "\"missing quote"})
+        self.assertGreater(code, 499)
```

Comment:
nit: Why not just check for code == 500?

Reply:
@pomegranited Same here - because there are a collection of 5xx codes :)

Reply:
👍